shapely/tests/test_constructive.py

```diff
@@ -105,12 +105,12 @@ def test_float_arg_nan(geometry, func):
 
 
 def test_buffer_cap_style_invalid():
-    with pytest.raises(ValueError, match="'invalid' is not a valid option"):
+    with pytest.raises(ValueError, match="'INVALID' is not a valid option"):
```

Comment:
This seems to point out an inconsistency: we instruct users to provide these in lowercase strings, which we automatically uppercase before checking against the enums.  `ParamEnum` handles the validation, but has no idea that the inputs are originally lowercase.


While minor, this might be confusing to users, since it returns a different casing on exception than they provided on input.

Perhaps it would be better if the entries were lowercase?

(note: the other enums `BinaryPredicate` and `DecodingErrorOptions` have lowercase entries)

Reply:
Back in these comments we decided to uppercase the enums since I believe that's the python style guide.
- https://github.com/shapely/shapely/pull/1508#issuecomment-1229214042
- https://github.com/shapely/shapely/pull/1508#issuecomment-1229277562

We could change things to use regular Enums with the values as the lowercase strings. (Though this would drop support for using 1/2/3 as inputs.)

```
class BufferCapStyle(Enum):
    ROUND = "round"
    FLAT = "flat"
    SQUARE = "square"


class BufferJoinStyle(Enum):
    ROUND = "round"
    MITRE = "mitre"
    BEVEL = "bevel"

>>> BufferCapStyle['ROUND']
<BufferCapStyle.ROUND: 'round'>
```

Then run input validation in `buffer` like
```
try:
   buffer_cap_style = BufferCapStyle[input.upper()]
except KeyError:
   raise ValueError(f"'{input"}' not in BufferCapStyle, options: {[e.value for e in BufferCapStyle]}")
```

which would return
```
'invalid' not in BufferCapStyle, options: ['round', 'flat', 'square']
```


Could also just do the input validation in `buffer` with the existing ParamEnum. Or could return the entries back to lowercase. I don't have a strong preference.

Reply:
After writing that all out I checked ParamEnum and realized it already does that validation that prints the enum options, the test just only checked the first half. The rest of that output would be `'Invalid' is not a valid option, must be one of 'SQUARE', 'ROUND', 'FLAT'`. That is kind of confusing.

Reply:
I think in this case we could opt for lowercase enum entries and deviate a bit from the Python style guide; that keeps us consistent with the other enums here, allows us to continue leveraging the validation messages from `ParamEnum` with no changes, and does not require a casing change to use incoming values.

Reply:
Changing back to lower case enum values is OK for me. 
Alternatively, we could also convert the values to lower case when printing them in the error message.

Reply:
I changed the enums to lower case, changing the case of the input to lowercase may be confusing as well.